sdk/core/core-client/src/interfaces.ts

```diff
@@ -413,36 +491,88 @@ export interface SimpleMapperType {
     | "any";
 }
 
+/**
+ * Helps build a mapper that describes how to map a set of properties of an object based on other mappers.
+ *
+ * Only one of the following properties should be present: `className`, `modelProperties` and `additionalProperties`.
+ */
 export interface CompositeMapperType {
+  /**
+   * Name of the composite mapper type.
+   */
   name: "Composite";
 
-  // Only one of the two below properties should be present.
-  // Use className to reference another type definition,
-  // and use modelProperties/additionalProperties when the reference to the other type has been resolved.
+  /**
+   * Use `className` to reference another type definition.
+   */
   className?: string;
 
+  /**
+   * Use `modelProperties` when the reference to the other type has been resolved.
+   */
   modelProperties?: { [propertyName: string]: Mapper };
+
+  /**
+   * Use `additionalProperties` when the reference to the other type has been resolved.
```

Comment:
if I'm not mistaken, this is used when a model has 'additionalProperties: true' and allows us to generically process unnamed model properties that are on the response object, e.g. 
```ts
interface SomeResponseModel {
  namedProp: string;
  otherNamedProp: number;
  // these are the additional properties that aren't explicitly defined
  [key: string]: string | number;
}
```

Reply:
I will push this `Used when a model has 'additionalProperties: true'. Allows the generic processing of unnamed model properties on the response object.` — feedback appreciated!